src/linter/ui5Types/fix/collections/sapUiCoreFixes.ts

```diff
@@ -112,9 +114,334 @@ t.declareModule("sap/ui/core/Core", [
 				return `${moduleIdentifier}.setTheme(${arg1})`;
 			},
 		})),
+		// Note that alternative replacement Component.get is meanwhile deprecated, too
 		t.method("getComponent", accessExpressionFix({
 			moduleName: "sap/ui/core/Component",
 			propertyAccess: "getComponentById",
 		})),
+		// Individual arguments must be mapped to "options" object
+		// The new API has no sync loading option, replacement is only safe when the options contain async:true
+		t.method("loadLibrary", callExpressionGeneratorFix({
+			moduleName: "sap/ui/core/Lib",
+			propertyAccess: "load",
+			validateArguments: (ctx: {json?: Record<string, string>}, _fixHints, arg1, arg2) => {
+				ctx.json = {};
+				if (arg2?.kind === SyntaxKind.ObjectLiteralExpression) {
+					let asyncOption = false;
+					const allowlistProps = ["url", "name"];
+					ts.forEachChild(arg2, function (node: ts.Node) {
+						if (ts.isPropertyAssignment(node) && ts.isIdentifier(node.name) && node.name.text === "async") {
+							if (node.initializer.kind === SyntaxKind.TrueKeyword) {
+								asyncOption = true; // Migration is possible, async loading is enabled
+							}
+						}
+
+						if (ts.isPropertyAssignment(node) && ts.isIdentifier(node.name)) {
+							const name = node.name.text;
+							if (allowlistProps.includes(name) && ts.isStringLiteralLike(node.initializer)) {
+								ctx.json![name] = node.initializer.text;
```

Comment:
See my comment below, maybe we can preserve trivia here as well.

```suggestion
								ctx.json![name] = node.initializer.getFullText();
```